File: truss-compiler-maven-plugin/src/main/java/truss/compiler/generator/syntax/xml/SyntaxClass.java
package truss.compiler.generator.syntax.xml;

import javax.xml.bind.annotation.*;
import java.util.ArrayList;
import java.util.List;

@XmlRootElement(name = "class")
@XmlAccessorType(XmlAccessType.FIELD)
public class SyntaxClass {
    @XmlAttribute
    private String name;
    @XmlAttribute
    private String base;
    @XmlAttribute(name = "abstract")
    private boolean abstract_;
    @XmlAttribute
    private boolean ignore;
    @XmlElement(name = "property")
    private List<SyntaxProperty> properties = new ArrayList<>();
    private String validation;
    private String members;

    public String getName() {
        return name;
    }

    public String getBase() {
        return base;
    }

    public boolean isAbstract() {
        return abstract_;
    }

    public boolean isIgnore() {
        return ignore;
    }

    public List<SyntaxProperty> getProperties() {
        return properties;
    }

    public String getValidation() {
        return validation;
    }

    public String getMembers() {
        return members;
    }
}

File: Truss.Compiler.Generate/Syntax/Xml/SyntaxClass.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml.Serialization;

namespace Truss.Compiler.Generate.Syntax.Xml {
    [XmlRoot("class")]
    public class SyntaxClass {
        [XmlAttribute("name")]
        public string Name { get; set; }

        [XmlAttribute("base")]
        public string Base { get; set; }

        [XmlAttribute("abstract")]
        public bool IsAbstract { get; set; }

        [XmlAttribute("ignore")]
        public bool IsIgnore { get; set; }

        [XmlElement("property")]
        public List<SyntaxProperty> Properties { get; set; }

        [XmlElement("validation")]
        public string Validation { get; set; }

        [XmlElement("members")]
        public string Members { get; set; }
    }
}
